lib/archethic/beacon_chain/subset/summary_cache.ex

```diff
@@ -27,18 +31,62 @@ defmodule Archethic.BeaconChain.Subset.SummaryCache do
       read_concurrency: true
     ])
 
-    :ok = recover_slots()
+    :ok = recover_slots(SummaryTimer.next_summary(DateTime.utc_now()))
+
+    PubSub.register_to_current_epoch_of_slot_time()
 
     {:ok, %{}}
   end
 
+  def handle_info({:current_epoch_of_slot_timer, slot_time}, state) do
+    # Check if the slot in the first one of the summary interval
+    previous_summary_time = SummaryTimer.previous_summary(slot_time)
+    first_slot_time = SlotTimer.next_slot(previous_summary_time)
+
+    if slot_time == first_slot_time do
+      Enum.each(
+        BeaconChain.list_subsets(),
+        &clean_previous_summary_cache(&1, previous_summary_time)
+      )
+
+      File.rm(recover_path(previous_summary_time))
+    end
+
+    {:noreply, state}
+  end
+
+  def code_change("1.0.7", state, _extra) do
+    next_summary_time = SummaryTimer.next_summary(DateTime.utc_now())
+    File.rename("slot_backup", "slot_backup-#{DateTime.to_unix(next_summary_time)}")
```

Comment:
I think `PubSub.register_to_current_epoch_of_slot_time()` added in the init should also be in the code change.

Reply:
Yes you're right !

Reply:
Thanks